cf-testd/cf-testd.c

```diff
@@ -676,11 +680,14 @@ static char *IncrementIPaddress(const char *ip_str)
         step = 3;
     }
 
-    uint32_t ip_num = ntohl(ip.s_addr);
+    uint32_t ip_num = ntohl(ip);
     ip_num += step;
-    ip.s_addr = htonl(ip_num);
+    ip = htonl(ip_num);
+
+    struct in_addr ip_struct;
+    ip_struct.s_addr = ip;
 
-    return xstrdup(inet_ntoa(ip));
+    return xstrdup(inet_ntoa(ip_struct));
```

Comment:
It's interesting how `inet_ntoa()` returns a non-const `char *` to a statically allocated buffer.

Reply:
Yeah, but it doesn't say you cannot modify it. Just that a next call will override it. I'm more concerned by the fact that this function cannot be thread-safe then.